src/boot/bootctl-uki.c

```diff
@@ -143,21 +144,51 @@ static int read_pe_section(FILE *uki, const struct PeSectionHeader *section,
         return 0;
 }
 
+static void inspect_osrel(char *osrel, size_t osrel_size) {
+        _cleanup_fclose_ FILE *s = NULL;
+        _cleanup_free_ char *pname = NULL, *name = NULL;
+        int r;
+
+        assert(osrel);
+        s = fmemopen(osrel, osrel_size, "r");
+        if (!s) {
+                log_warning_errno(errno, "Failed to open embedded os-release file, ignoring: %m");
+                return;
+        }
```

Comment:
you could do:

```c
if (!s)
        return (void) log_warning_errno(errno, "…");
```

we usually do that in cases like thi.

Reply:
Neat trick, didn't know that one.